sdk/communication/Azure.Communication.MediaComposition/tests/samples/Sample1_MediaCompositionClient.cs

```diff
@@ -0,0 +1,157 @@
+                        call: "teamsMeeting")
+                        {
+                            PlaceholderImageUri = "https://imageendpoint"
+                        }
+                    }
+                },
+                 {
+                    "jane",
+                    new MediaInput() {
+                        Participant = new ParticipantInput(
+                        id: new CommunicationIdentifierModel() { MicrosoftTeamsUser = new MicrosoftTeamsUserIdentifierModel("2dd69470-dc25-49cf-b5c3-f562f08bf3b2") },
+                        call: "teamsMeeting")
+                        {
+                            PlaceholderImageUri = "https://imageendpoint"
+                        }
+                    }
+                },
+                {
+                    "jerry",
+                    new MediaInput() {
+                        Participant = new ParticipantInput(
+                        id: new CommunicationIdentifierModel() { MicrosoftTeamsUser = new MicrosoftTeamsUserIdentifierModel("30e29fde-ac1c-448f-bb34-0f3448d5a677") },
+                        call: "teamsMeeting")
+                        {
+                            PlaceholderImageUri = "https://imageendpoint"
+                        }
+                    }
+                },
+                {
+                    "teamsMeeting",
+                    new MediaInput()
+                    {
+                        TeamsMeeting = new TeamsMeeting("https://teamsJoinUrl")
+                    }
+                }
+            };
+
+            var outputs = new Dictionary<string, MediaOutput>()
+            {
+                {
+                    "acsGroupCall",
+                    new MediaOutput()
+                    {
+                        GroupCall = new GroupCall("d12d2277-ffec-4e22-9979-8c0d8c13d191")
+                    }
+                }
+            };
+            await mediaCompositionClient.CreateAsync(mediaCompositionId, layout, inputs, outputs);
+        }
+
+        public async Task GetMediaCompositionAsync()
+        {
+            var mediaCompositionClient = CreateMediaCompositionClient();
+            var gridMediaComposition = await mediaCompositionClient.GetAsync(mediaCompositionId);
+        }
+
+        public async Task UpdateLayoutMediaCompositionAsync()
+        {
+            var mediaCompositionClient = CreateMediaCompositionClient();
+            var layout = new MediaCompositionLayout()
+            {
+                Resolution = new LayoutResolution(720, 480),
+                Presenter = new PresenterLayoutOptions("jill", "jack")
+                {
+                    SupportPosition = SupportPosition.BottomRight,
+                    SupportAspectRatio = 3/2
+                }
+            };
+            await mediaCompositionClient.UpdateAsync(mediaCompositionId, layout);
```

Comment:
We'll need more discussions whether we want to support a full Update or instead have smaller methods like `AddSource`. The reason is that merge patch semantics are hard to convey, and it's not clear that passing a dictionary will only add new items but not remove pre-existing ones, or that one has to set a dictionary item to null to remove it.

Azure SDK is actually changing their engineering strategy to generate the data-plane client entirely and only start with what they call "protocol"-based methods instead of "model"-based methods. The idea is that a team can start with the protocol layer where developers send raw requests, and can then add model-based methods later. So far we haven't adopted that new approach in ACS yet, but maybe we can see if it would make sense for media composition?

https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/core/Azure.Core/samples/ProtocolMethods.md